share/completions/git.fish

```diff
@@ -549,7 +549,15 @@ function __fish_git_needs_command
 end
 
 function __fish_git_config_keys
-    git config -l | string match -r '[^=]+'
+    # Print already defined config values first
+    # Config keys may span multiple lines, so parse using null char
+    # With -z, key and value are separated by space, not "="
+    git config -lz | while read -lz key value
+        # Print only first line of value(with an ellipsis) if multiline
+        printf '%s\t%s\n' $key (string replace \n …\n $value)[1]
+    end
+    # Print all recognized config keys; duplicates are not shown twice by fish
+    printf '%s\n' (__fish_git help --config)[1..-2] # Last line is a footer; ignore it
```

Comment:
For me the last line is a footer and the line before it is *empty*, so I'd assume this needs a `-3`?

Reply:
I did notice that the second last line in blank, but the extra blank line seemed to be trimmed and I didn't give it anymore thought. I'll change the range to `[1..-3]` then. Is there any reason other than clarity though ?

Reply:
Oh, right, we trim blank entries already.

No, that's okay then and the change isn't needed.